db/schema.sql

```diff
@@ -0,0 +1,103 @@
+SELECT count(*) FROM information_schema.tables WHERE table_schema = CURRENT_SCHEMA() AND table_name = 'migrations' AND table_type = 'BASE TABLE';
+
+CREATE TABLE "migrations" (
+    "id" varchar(255),
+    PRIMARY KEY ("id")
+);
+
+CREATE TABLE "tenants" (
+    "created_at" timestamptz NOT NULL,
+    "updated_at" timestamptz,
+    "deleted_at" timestamptz,
+    "tenant_id" serial,
+    "tenant_storage_contract_cid" text NOT NULL,
+    "tenant_meta" jsonb NOT NULL DEFAULT '{}',
+    "tenant_settings" jsonb NOT NULL DEFAULT '{}',
+    PRIMARY KEY ("tenant_id")
+);
+
+CREATE INDEX IF NOT EXISTS "idx_tenants_deleted_at" ON "tenants" ("deleted_at");
+
+CREATE TABLE "tenants_sps" (
+    "tenant_id" integer,
+    "sp_id" integer,
+    "tenant_sp_state" tenant_sp_state NOT NULL DEFAULT 'eligible',
+    "tenant_sps_meta" jsonb NOT NULL DEFAULT '{}',
+    PRIMARY KEY ("tenant_id", "sp_id")
+);
+
+CREATE TABLE "addresses" (
+    "created_at" timestamptz NOT NULL,
+    "updated_at" timestamptz,
+    "deleted_at" timestamptz,
+    "tenant_id" integer NOT NULL,
+    "address" text NOT NULL,
+    "address_actor_id" bigint,
+    "address_is_signing" boolean NOT NULL DEFAULT true,
+    CONSTRAINT "fk_tenants_tenant_addresses" FOREIGN KEY ("tenant_id") REFERENCES "tenants"("tenant_id")
+);
+
+CREATE INDEX IF NOT EXISTS "idx_addresses_deleted_at" ON "addresses" ("deleted_at");
+CREATE UNIQUE INDEX IF NOT EXISTS "idx_address_tenant_id" ON "addresses" ("tenant_id", "address");
+
+CREATE TABLE "tenant_sp_eligibility_clauses" (
+    "created_at" timestamptz NOT NULL,
+    "updated_at" timestamptz,
+    "deleted_at" timestamptz,
+    "tenant_id" integer,
+    "clause_attribute" text NOT NULL,
+    "clause_operator" comparison_operator NOT NULL,
+    "clause_value" text NOT NULL,
+    CONSTRAINT "fk_tenants_tenant_sp_eligibility" FOREIGN KEY ("tenant_id") REFERENCES "tenants"("tenant_id")
+);
+
+CREATE INDEX IF NOT EXISTS "idx_tenant_sp_eligibility_clauses_deleted_at" ON "tenant_sp_eligibility_clauses" ("deleted_at");
+
+CREATE TABLE "collections" (
+    "created_at" timestamptz NOT NULL,
+    "updated_at" timestamptz,
+    "deleted_at" timestamptz,
+    "collection_id" serial,
+    "tenant_id" integer NOT NULL,
+    "collection_name" text NOT NULL,
+    "collection_active" boolean NOT NULL,
+    "collection_piece_source" jsonb NOT NULL DEFAULT '{}',
+    "collection_deal_params" jsonb NOT NULL DEFAULT '{}',
+    PRIMARY KEY ("collection_id"),
+    CONSTRAINT "fk_tenants_collections" FOREIGN KEY ("tenant_id") REFERENCES "tenants"("tenant_id")
+);
+
+CREATE INDEX IF NOT EXISTS "idx_collections_deleted_at" ON "collections" ("deleted_at");
+
+CREATE TABLE "labels" (
```

Comment:
This table has no unique constraint, is this desired?

Reply:
There are these two below - this should constrain it properly right?

```
CREATE UNIQUE INDEX IF NOT EXISTS "idx_label_tenant_id_label_text" ON "labels" ("tenant_id", "label_text");
CREATE UNIQUE INDEX IF NOT EXISTS "idx_label_tenant_id_label_id" ON "labels" ("tenant_id", "label_id");
```